contracts/contracts/activities/Activity.sol

```diff
@@ -3,54 +3,70 @@ pragma solidity ^0.8.23;
-    /// If there is a configuration change applied at this checkpoint, this carries information about the _old_ validator set.
-    uint64 blocksCommitted;
-    /// @dev Other metadata
-    bytes metadata;
+// Compressed representation of the activity summary that can be embedded in checkpoints to propagate up the hierarchy.
+struct CompressedActivitySummary {
+    Consensus.Compressed consensus;
 }
 
-/// The summary for the child subnet activities that should be submitted to the parent subnet
-/// together with a bottom up checkpoint
-struct ActivitySummary {
-    /// The total number of distintive validators that have mined
-    uint64 totalActiveValidators;
-    /// The activity commitment for validators
-    bytes32 commitment;
+/// Namespace for consensus-level activity summaries.
+library Consensus {
+    // Aggregated stats for consensus-level activity.
+    struct Aggregated {
+        /// The total number of unique validators that have mined within this period.
+        uint64 totalActiveValidators;
+        /// The total number of blocks committed by all validators during this period.
+        uint64 totalNumBlocksCommitted;
+    }
 
-    // TODO: add relayed rewarder commitment
-}
+    // The full activity summary for consensus-level activity.
+    struct Full {
+        Aggregated aggregated;
+        /// The breakdown of activity per validator.
+        ValidatorDetail[] validatorDetails;
+    }
 
-/// The summary for a single validator
-struct ValidatorSummary {
-    /// @dev The child subnet checkpoint height associated with this summary
-    uint64 checkpointHeight;
-    /// @dev The validator whose activity we're reporting about.
-    address validator;
-    /// @dev The number of blocks committed by each validator in the position they appear in the validators array.
-    /// If there is a configuration change applied at this checkpoint, this carries information about the _old_ validator set.
-    uint64 blocksCommitted;
-    /// @dev Other metadata
-    bytes metadata;
+    // The compresed representation of the activity summary for consensus-level activity suitable for embedding in a checkpoint.
+    struct Compressed {
+        Aggregated aggregated;
+        /// The commitment for the validator details, so that we don't have to transmit them in full.
+        bytes32 commitment;
+    }
+
+    struct ValidatorDetail {
+        /// @dev The validator whose activity we're reporting about, identified by the Ethereum address corresponding
+        /// to its secp256k1 pubkey.
+        address validator;
+        /// @dev The number of blocks committed by this validator during the summarised period.
+        uint64 blocksCommitted;
+        /// @dev Other metadata
+        bytes metadata;
+    }
 }
 
 /// The proof required for validators to claim rewards
 struct ValidatorClaimProof {
-    ValidatorSummary summary;
+    uint64 checkpointHeight;
```

Comment:
@raulk Add in checkpoint height to identify which commitment to use

Reply:
I think this should be an unwrapped argument. It is not part of the proof per se, but rather an extra argument we need to pass to the method to identify the height against we're presenting the proof. For the batch claim, we should take an array of heights and an array of proofs.

Reply:
See https://github.com/consensus-shipyard/ipc/pull/1205#discussion_r1846530654.